ui/src/features/Editor/useCanvasCopyPaste.ts

```diff
@@ -71,18 +51,20 @@ export default ({
         );
       }
       return newEdges;
-    };
-
-    const newNodeCreation = (pn: Node[]): Node[] => {
+    },
+    [],
+  );
+  const newNodeCreation = useCallback(
+    (pn: Node[], pastedWorkflows?: Workflow[], isPaste?: boolean): Node[] => {
```

Comment:
`pastedNodes`
Also, isPasted is only used when pastedWorkflows exist, so you can remove that and then below do `if (newNode.type === "subworkflow" && pastedWorkflows) ....`